buffer/src/main/java/io/netty/buffer/AdaptivePoolingAllocator.java

```diff
@@ -243,7 +254,8 @@ ByteBuf allocate(int size, int maxCapacity) {
     private AdaptiveByteBuf allocate(int size, int maxCapacity, Thread currentThread, AdaptiveByteBuf buf) {
         AdaptiveByteBuf allocated = null;
         if (size <= MAX_POOLED_BUF_SIZE) {
-            int index = binarySearchInsertionPoint(Arrays.binarySearch(SIZE_CLASSES, size));
+            final int index = sizeClassIndexOf(size);
+            assert binarySearchInsertionPoint(Arrays.binarySearch(SIZE_CLASSES, size)) == index;
```

Comment:
This one is not super nice on tests wdyt @chrisvest it was my safe net in case I made some silly math mistake

Reply:
And you advice was on the spot - I made some silly math mistake :dagger:

Reply:
Added test, now running a microbench